src/Nancy/Diagnostics/DiagnosticsHook.cs

```diff
@@ -188,7 +188,7 @@ private static Response ExecuteDiagnostics(NancyContext ctx, IRouteResolver rout
 
             if (resolveResult.After != null)
             {
-                resolveResult.After.Invoke(ctx, CancellationToken);
+                resolveResult.After.Invoke(ctx, CancellationToken).Wait();

```

Comment:
This is one of those changes that will have a knock on effect and will explode this PR. I would agree to merge this PR as is and fix this `.Wait()` in a subsequent PR.